src/MongoDB/WriteResult.c

```diff
@@ -50,6 +50,108 @@ PHONGO_API zend_class_entry *php_phongo_writeresult_ce;
 
 zend_object_handlers php_phongo_handler_writeresult;
 
+static bool php_phongo_writeresult_get_writeconcernerror(php_phongo_writeresult_t *intern, zval *return_value TSRMLS_DC)
+{
+	bson_iter_t iter, child;
+#if PHP_VERSION_ID >= 70000
+	zval writeconcernerror;
+#else
+	zval *writeconcernerror = NULL;
+#endif
+
+	ZVAL_NULL(return_value);
+
+	if (bson_iter_init_find(&iter, intern->reply, "writeConcernErrors") && BSON_ITER_HOLDS_ARRAY(&iter) && bson_iter_recurse(&iter, &child)) {
+		while (bson_iter_next(&child)) {
+			bson_t cbson;
+			uint32_t len;
+			const uint8_t *data;
+
+			if (!BSON_ITER_HOLDS_DOCUMENT(&child)) {
+				continue;
+			}
+
+			bson_iter_document(&child, &len, &data);
+
+			if (!bson_init_static(&cbson, data, len)) {
+				continue;
+			}
+
+#if PHP_VERSION_ID >= 70000
+			if (!phongo_writeconcernerror_init(&writeconcernerror, &cbson TSRMLS_CC)) {
+				zval_ptr_dtor(&writeconcernerror);
+				return false;
+			}
+
+			ZVAL_ZVAL(return_value, &writeconcernerror, 1, 1);
+#else
+			MAKE_STD_ZVAL(writeconcernerror);
+
+			if (!phongo_writeconcernerror_init(writeconcernerror, &cbson TSRMLS_CC)) {
+				zval_ptr_dtor(&writeconcernerror);
+				return false;
+			}
+
+			ZVAL_ZVAL(return_value, writeconcernerror, 1, 1);
```

Comment:
I ran into trouble trying to use `zval_ptr_dtor()` on the input argument so it seemed more straightforward to create my own zval in this function and copy it into the return value on success. For any failure case, we clean up our internally allocated zval and leave the return value initialized to null.